topi/include/topi/transform.h

```diff
@@ -575,5 +575,50 @@ inline Tensor take(const Tensor& a,
         }, name, tag);
 }
 
+/*!
+* \brief Return the elements, either from x or y, depending on the condition.
+*
+* \param condition The condition array.
+* \param x First array to be selected.
+* \param y Second array to be selected.
+* \param name The name of the operation.
+* \param tag The tag to mark the operation.
+*
+* \return A Tensor selected from x or y depending on condition.
+*/
+inline Tensor where(const Tensor& condition,
+                    const Tensor& x,
+                    const Tensor& y,
+                    std::string name = "tensor",
+                    std::string tag = kInjective) {
+  CHECK_EQ(x->shape.size(), y->shape.size())
+    << "x and y must have the same shape.Got different number of dimension: "
+    << x->shape.size() << " vs " << y->shape.size();
+  Array<Expr> oshape = x->shape;
+  Tensor out;
+
+  if (condition->shape.size() != 1) {
+    CHECK_EQ(condition->shape.size(), x->shape.size())
+      << "condition and x must have the same shape.Got "
+         "different number of dimension: "
+      << condition->shape.size() << " vs " << x->shape.size();
+    out = compute(
+      oshape, [&](const Array<Var>& indices) {
+        return tvm::select(condition(indices) != 0, x(indices), y(indices));
+      }, name, tag);
+  } else {
+    CHECK_EQ(condition->shape.size(), 1) << "condition array must be either "
+        "have the same shape as x or to be a 1-D array.";
+    out = compute(
+      oshape, [&](const Array<Var>& indices) {
+        Array<Expr> condition_idx{indices[0]};
```

Comment:
Is condition index always from the first dimension? Is it where operator's property or user can specify the axis?

Reply:
There are two cases for condition shape. 1. the same as x. 2. to be 1-D. This is for 1-D case.

Reply:
In second case of 1-D condition array, 'condition_idx{indices[0]}' always take first entry of indices array. It can be done 'condition_idx{indices[axis]}' where user can specify the axis along which output is taken from x or y (example: in 2D x and y, choose complete column of x or row of x if condition array has true entry. Current implementation always takes column of x if condition is true.). 

But never mind, i checked TF and MxNet implementation, both are using first dimension. This looks okay.